nautobot/extras/models/groups.py

```diff
@@ -431,6 +431,14 @@ def clean_filter(self):
         if not filterset.is_valid():
             raise ValidationError(filterset.errors)
 
+    def delete(self):
+        """Check if we're a parent or child and attempt to block delete if we are."""
+        if self.children.exists():
+            raise ValidationError({"children": "Cannot delete DynamicGroup while parent of other DynamicGroups."})
```

Comment:
Why can't we delete an aggregate group? I get blocking on deletion of a child group, since this would result in a change to the parent group, but not the other way around.

Reply:
Yeah I'll just make it child not parent.